### PR TITLE
Add test cases for both Solr/ES with {Core18, Robust04, and MS MARCO passage}

### PR DESCRIPTION
**Pull request by x389liu**
Ref: Issue #927

Note that the current AP values of Solr with MS MARCO passage and ES with Core18 do not match their expected values. All other cases should work well.

Specifically: 
- Solr with MS MARCO passage produces `[FAILED] 0.187 MAP, expected 0.1956 MAP!`
- ES with Core18 produces `[FAILED] 0.2401 MAP, expected 0.2495 MAP!`

### CODE REVIEW

**Comment by lintool** [comment]
@x389liu Why can't we do this? Abstract the query generator functionality into something like `SolrBagOfWordsQueryGenerator`, which should exhibit the same behavior as `BagOfWordsQueryGenerator`. Then you can easily unit test it.

The topics are already coded here: https://github.com/castorini/anserini/blob/master/src/main/java/io/anserini/search/topicreader/Topics.java

You can run through `BagOfWordsQueryGenerator`, print out the query, and make sure they exactly match the Solr version?

**Comment by lintool** [comment]
Alternatively, could we generate the query with `BagOfWordsQueryGenerator`, dump out the query in the proper Lucene format, and then feed into `SolrQuery`?

**Comment by lintool** [comment]
Take a look at this: https://lucene.apache.org/solr/guide/8_3/overview-of-searching-in-solr.html
Offers a pretty good explanation of query parsers, etc. - should be enough to help you debug.

**Comment by x389liu** [comment]
After spending a while on debugging the inconsistent AP values problem, the MS MARCO case is resolved that it works well on Solr now. But Core18 still does not work as expected on Elasticsearch. So we decided to include only working test cases (i.e. Robust04 and MS MARCO on Solr) for now.

File: src/main/java/io/anserini/search/SearchSolr.java [hunk 1]
```diff
@@ -198,11 +198,18 @@ public SearchSolr(Args args) throws IOException {
 
     SolrDocumentList results = null;
 
+    // Replace some characters in query which are special syntax in Solr query parser
```
Comment (lintool) [inline src/main/java/io/anserini/search/SearchSolr.java]: Why can't this be replaced with a single regular expression?

Comment (x389liu) [reply src/main/java/io/anserini/search/SearchSolr.java]: Thanks for the advice! Updated.

**Comment by lintool** [comment]
This PR title is misnamed, but I confirm that I am able to replicate the MS MARCO results in Solr. So, for Solr, Robust04, Core18, and MS MARCO passage all work. 

Merging.

**Comment by lintool** [comment]
Closed by #1030

**Pull request merged**